src/main/daml/Daml/Finance/Interface/Types/Date/RollConvention.daml

```diff
@@ -24,10 +24,22 @@ data PeriodEnum
   -- ^ Year
   deriving (Eq, Show)
 
+-- | An enum type to either specify a period or a single term.
+data ScheduleFrequencyEnum
```

Comment:
All right, I just realised that this is no longer an `Enum` once we add the `periodMultiplier` in there 😓 

Also, it would be nice to avoid having to define a `RollConvention` for a `SinglePeriod` schedule, as that is not well-defined.

I looked at these types for a bit and have just committed a proposal to your branch. Let me know if you think it makes sense, otherwise i will remove it.

Reply:
Cool. We could even get rid of the `Frequency` and add the `RollConvention` into a period. This change is a bit bigger though, so I can come up with a proposal on a separate PR and we can see if it makes sense

Reply:
Sounds good!